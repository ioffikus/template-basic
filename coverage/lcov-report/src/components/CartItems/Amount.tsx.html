<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for src/components/CartItems/Amount.tsx</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../../../prettify.css" />
    <link rel="stylesheet" href="../../../base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../../../sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="../../../index.html">All files</a> / <a href="index.html">src/components/CartItems</a> Amount.tsx
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">96.88% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>31/32</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">75% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>12/16</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">90.91% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>10/11</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">96.77% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>30/31</span>
      </div>
    </div>
  </div>
  <div class='status-line high'></div>
<pre><table class="coverage">
<tr><td class="line-count quiet">1
2
3
4
5
6
7
8
9
10
11
12
13
14
15
16
17
18
19
20
21
22
23
24
25
26
27
28
29
30
31
32
33
34
35
36
37
38
39
40
41
42
43
44
45
46
47
48
49
50
51
52
53
54
55
56
57
58
59
60
61
62
63
64
65
66
67
68
69
70
71
72
73
74
75
76
77
78
79
80
81
82
83
84
85
86
87
88
89
90
91
92
93
94
95
96
97
98
99
100
101
102
103
104
105
106
107
108
109
110
111
112
113
114
115
116
117
118
119
120
121
122
123
124
125
126
127
128
129
130
131
132
133
134
135
136
137
138
139
140
141
142
143
144
145
146
147
148
149
150
151
152
153
154
155
156
157
158
159
160
161
162</td><td class="line-coverage quiet"><span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">11x</span>
<span class="cline-any cline-yes">11x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">8x</span>
<span class="cline-any cline-yes">8x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">8x</span>
<span class="cline-any cline-yes">8x</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">8x</span>
<span class="cline-any cline-yes">3x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">8x</span>
<span class="cline-any cline-yes">3x</span>
<span class="cline-any cline-yes">3x</span>
<span class="cline-any cline-yes">3x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">3x</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">8x</span>
<span class="cline-any cline-yes">8x</span>
<span class="cline-any cline-yes">8x</span>
<span class="cline-any cline-yes">8x</span>
<span class="cline-any cline-yes">8x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">7x</span>
<span class="cline-any cline-yes">3x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">12x</span>
<span class="cline-any cline-yes">12x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span></td><td class="text"><pre class="prettyprint lang-js">import React from 'react';
import { connect } from 'react-redux';
import { InputNumber, Spin } from 'antd';
import dispatch from 'src/store/dispatch';
import { increaseQuantity, decreaseQuantity } from 'src/ducks/cart';
&nbsp;
interface IOwnProps {
  mayChangeAmountItems: boolean;
  uid: string;
  count: number;
  pendingShowSpinning?: number;
  pendingHideSpinning?: number;
}
&nbsp;
interface IConnectedState {}
&nbsp;
type IProps = IOwnProps &amp; IConnectedState;
&nbsp;
interface IState {
  isSpinning: boolean;
}
&nbsp;
const mapStateTo<span class="missing-if-branch" title="else path not taken" >E</span>Props = (state: Store.IState) =&gt; ({});
&nbsp;
const WrappedAmount = connect&lt;IConnectedState, undefined, IOwnProps&gt;(mapStateToProps)(
  class Amount extends React.Component&lt;IProps, IState&gt; {
    static defaultProps = {
      pendingShowSpinning: 400,
      pendingHideSpinning: 300,
    };
&nbsp;
    spinningTimeoutId: number;
&nbsp;
    state: IStat<span class="missing-if-branch" title="else path not taken" >E</span>e = {
      isSpinning: false,
    };
&nbsp;
    showSpinning = () =&gt; {<span class="fstat-no" title="function not covered" ></span>
      if (!!this.spinningTim<span class="cstat-no" title="statement not covered" >eoutId) {</span>
        window.clearTimeout(this.spinningTimeoutId);
        this.spinningTimeoutId = null;
      }
      this.spinningTimeoutId = window.setTimeout(() =&gt; {
        this.setState({ isSpinning: true });
      }, this.props.pendingShowSpinning);
    };
&nbsp;
    hideSpinning = () =&gt; {
      if (!!this.spinningTimeoutId) {
        window.clearTimeout(this.spinningTimeoutId);
        this<span class="missing-if-branch" title="else path not taken" >E</span>.spinningTimeoutId = null;
      }<span class="missing-if-branch" title="else path not taken" >E</span>
      if (this.state.isSpinning) {
        this.spinningTimeoutId = window.setTimeout(() =&gt; {
          this.setState({ isSpinning: false });
        }, this.props.pendingHideSpinning);
      }
    };
&nbsp;
    handleChangeAmount = (value: number) =&gt; {
      if (process.env.BROWSER) {
        const { uid, count } = this.props;
        this.showSpinning();
        if (count &lt; value) {
          dispatch(
            increaseQuantity(uid, () =&gt; {
              this.hideSpinning();
            }),
          );
        } else {
          dispatch(
            decreaseQuantity(uid, () =&gt; {
              this.hideSpinning();
            }),
          );
        }
      }
    };
&nbsp;
    componentWillUnmount() {
      if (process.env.BROWSER) {
        if (!!this.spinningTimeoutId) {
          window.clearTimeout(this.spinningTimeoutId);
        }
      }
    }
&nbsp;
    render() {
      const { mayChangeAmountItems, count } = this.props;
      return (
        &lt;div className="root"&gt;
          &lt;style jsx&gt;{`
            .root {
              display: inline-block;
              vertical-align: top;
            }
            .root :global(.ant-input-number) {
              border: 0;
              display: flex;
              justify-content: center;
              background: transparent;
              color: #062f49;
            }
            .root :global(.ant-input-number:hover) {
              border-color: transparent;
            }
            .root :global(.ant-input-number-focused) {
              box-shadow: none;
              border-color: transparent;
            }
            .root :global(.ant-input-number-handler-wrap) {
              border: 0;
              opacity: 1;
              width: 100%;
              display: flex;
              flex-direction: row-reverse;
              background: transparent;
            }
            .root :global(.ant-input-number-input-wrap) {
              position: relative;
              z-index: 3;
              width: 32px;
              height: 32px;
            }
            .root :global(.ant-input-number-handler) {
              width: 50%;
              height: 100%;
              border: 0;
              top: 0;
              bottom: 0;
              background: transparent;
            }
            .root :global(.ant-input-number-handler:hover) {
              height: 100% !important;
            }
            .root :global(.ant-input-number-handler-down-inner) {
              right: auto;
              left: 4px;
            }
            .root :global(.ant-input-number-input) {
              background-color: #e5ecf4;
              border-radius: 8px;
              color: inherit;
              text-align: right;
              font-weight: bold;
            }
          `}&lt;/style&gt;
          {mayChangeAmountItems ? (
            &lt;Spin spinning={this.state.isSpinning}&gt;
              &lt;InputNumber min={1} value={count} onChange={this.handleChangeAmount} /&gt;
            &lt;/Spin&gt;
          ) : (
            count
          )}
        &lt;/div&gt;
      );
    }
  },
);
&nbsp;
export default WrappedAmount;
&nbsp;</pre></td></tr>
</table></pre>
<div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Fri Dec 29 2017 11:52:09 GMT+0300 (MSK)
</div>
</div>
<script src="../../../prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="../../../sorter.js"></script>
</body>
</html>
