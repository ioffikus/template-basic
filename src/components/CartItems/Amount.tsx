import React from 'react';
import { connect } from 'react-redux';
import { InputNumber, Spin } from 'antd';
import dispatch from 'src/store/dispatch';
import { increaseQuantity, decreaseQuantity } from 'src/ducks/cart';

interface IOwnProps {
  mayChangeAmountItems: boolean;
  uid: string;
  count: number;
  pendingShowSpinning?: number;
  pendingHideSpinning?: number;
}

interface IConnectedState {}

type IProps = IOwnProps & IConnectedState;

interface IState {
  isSpinning: boolean;
}

const mapStateToProps = (state: Store.IState) => ({});

const WrappedAmount = connect<IConnectedState, undefined, IOwnProps>(mapStateToProps)(
  class Amount extends React.Component<IProps, IState> {
    static defaultProps = {
      pendingShowSpinning: 400,
      pendingHideSpinning: 300,
    };

    spinningTimeoutId: number;

    state: IState = {
      isSpinning: false,
    };

    showSpinning = () => {
      if (!!this.spinningTimeoutId) {
        window.clearTimeout(this.spinningTimeoutId);
        this.spinningTimeoutId = null;
      }
      this.spinningTimeoutId = window.setTimeout(() => {
        this.setState({ isSpinning: true });
      }, this.props.pendingShowSpinning);
    };

    hideSpinning = () => {
      if (!!this.spinningTimeoutId) {
        window.clearTimeout(this.spinningTimeoutId);
        this.spinningTimeoutId = null;
      }
      if (this.state.isSpinning) {
        this.spinningTimeoutId = window.setTimeout(() => {
          this.setState({ isSpinning: false });
        }, this.props.pendingHideSpinning);
      }
    };

    handleChangeAmount = (value: number) => {
      if (process.env.BROWSER) {
        const { uid, count } = this.props;
        this.showSpinning();
        if (count < value) {
          dispatch(
            increaseQuantity(uid, () => {
              this.hideSpinning();
            }),
          );
        } else {
          dispatch(
            decreaseQuantity(uid, () => {
              this.hideSpinning();
            }),
          );
        }
      }
    };

    componentWillUnmount() {
      if (process.env.BROWSER) {
        if (!!this.spinningTimeoutId) {
          window.clearTimeout(this.spinningTimeoutId);
        }
      }
    }

    render() {
      const { mayChangeAmountItems, count } = this.props;
      return (
        <div className="root">
          <style jsx>{`
            .root {
              display: inline-block;
              vertical-align: top;
            }
            .root :global(.ant-input-number) {
              border: 0;
              display: flex;
              justify-content: center;
              background: transparent;
              color: #062f49;
            }
            .root :global(.ant-input-number:hover) {
              border-color: transparent;
            }
            .root :global(.ant-input-number-focused) {
              box-shadow: none;
              border-color: transparent;
            }
            .root :global(.ant-input-number-handler-wrap) {
              border: 0;
              opacity: 1;
              width: 100%;
              display: flex;
              flex-direction: row-reverse;
              background: transparent;
            }
            .root :global(.ant-input-number-input-wrap) {
              position: relative;
              z-index: 3;
              width: 32px;
              height: 32px;
            }
            .root :global(.ant-input-number-handler) {
              width: 50%;
              height: 100%;
              border: 0;
              top: 0;
              bottom: 0;
              background: transparent;
            }
            .root :global(.ant-input-number-handler:hover) {
              height: 100% !important;
            }
            .root :global(.ant-input-number-handler-down-inner) {
              right: auto;
              left: 4px;
            }
            .root :global(.ant-input-number-input) {
              background-color: #e5ecf4;
              border-radius: 8px;
              color: inherit;
              text-align: right;
              font-weight: bold;
            }
          `}</style>
          {mayChangeAmountItems ? (
            <Spin spinning={this.state.isSpinning}>
              <InputNumber min={1} value={count} onChange={this.handleChangeAmount} />
            </Spin>
          ) : (
            count
          )}
        </div>
      );
    }
  },
);

export default WrappedAmount;
