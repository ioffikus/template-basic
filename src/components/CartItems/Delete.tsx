import React from 'react';
import { connect } from 'react-redux';
import { Button } from 'antd';
import dispatch from 'src/store/dispatch';
import { deleteFromCart } from 'src/ducks/cart';
import Svg from 'src/components/Svg';

interface IOwnProps {
  mayDeleteItems: boolean;
  uid: string;
}

interface IConnectedState {}

type IProps = IOwnProps & IConnectedState;

interface IState {}

const mapStateToProps = (state: Store.IState) => ({});

const WrappedDelete = connect<IConnectedState, undefined, IOwnProps>(mapStateToProps)(
  class Delete extends React.Component<IProps, IState> {
    handleDelete = (uid: string) => () => {
      dispatch(deleteFromCart(uid));
    };

    render() {
      const { mayDeleteItems, uid } = this.props;
      return (
        <div className="root">
          <style jsx>{`
            .root {
              text-align: right;
            }
            .root :global(.button-delete) {
              background: transparent;
              color: #b2bdcb;
            }
            .root :global(.icon-svg-trash) {
              height: 15px;
              width: 14px;
            }
          `}</style>
          {mayDeleteItems ? (
            <Button shape="circle" onClick={this.handleDelete(uid)} className="button-delete">
              <Svg name="trash" className="icon-svg-trash" />
            </Button>
          ) : (
            <span />
          )}
        </div>
      );
    }
  },
);

export default WrappedDelete;
