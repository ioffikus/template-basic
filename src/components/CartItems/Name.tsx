import React from 'react';
import { Link } from 'react-router-dom';

interface IProps {
  uid: string;
  shortDescription: string;
  name: string;
}

interface IState {}

class Copyright extends React.Component<IProps, IState> {
  render() {
    const { uid, shortDescription, name } = this.props;
    return (
      <div className="root">
        <style jsx>{`
          .root {
            font-size: 16px;
            line-height: 1.3;
            font-weight: bold;
            overflow: hidden;
            text-overflow: ellipsis;
            white-space: nowrap;
          }

          .root :global(.name) {
            color: inherit;
            white-space: nowrap;
          }
          .root :global(.code) {
            color: #b2bdcb;
            font-size: 12px;
          }
        `}</style>
        <Link to={`/product/${uid}`} className="name" title={shortDescription}>
          {name}
        </Link>
        <div className="code"># {uid}</div>
      </div>
    );
  }
}

export default Copyright;
