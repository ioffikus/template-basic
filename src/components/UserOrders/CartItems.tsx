import React from 'react';
import { Link } from 'react-router-dom';
import { Table } from 'antd';
import _ from 'lodash';
import Price from 'src/components/Price';
import CartProductImage from 'src/components/CartProductImage';

interface IRecord {
  key: number;
  pictureUrl: string;
  name: string;
  count: number;
  price: number;
  currency: string;
  uid: string;
}

interface IProps {
  ui: Alicanto.Models.UI;
  items: Alicanto.Models.CatalogItem[];
}

interface IState {}

class CartItems extends React.Component<IProps, IState> {
  getDataSource = (): IRecord[] => {
    return this.props.items.map(item => {
      const key = item.id;
      const uid = item.uid;
      const schemaFields = item.schema_fields;
      const pictureUrl = _.get(schemaFields, 'picture_url.value', '');
      const name = _.get(schemaFields, 'name.value', '');
      const count = _.get(item, 'count', 0);
      const currency = _.get(schemaFields, 'currency.value', '');
      const price = _.get(schemaFields, 'price.value', 0);
      return { key, pictureUrl, name, count, price, currency, uid };
    });
  };

  render() {
    const commonUI = this.props.ui.common;
    const columns = [
      {
        title: commonUI.tables.cells.image.i18n.defaultText,
        key: 'image',
        render: (text: string, record: IRecord) => {
          return <CartProductImage pictureUrl={record.pictureUrl} imageAlt={record.name} />;
        },
      },
      {
        title: commonUI.tables.cells.name.i18n.defaultText,
        key: 'name',
        render: (text: string, record: IRecord) => {
          return <Link to={`/product/${record.uid}`}>{record.name}</Link>;
        },
      },
      { title: commonUI.tables.cells.count.i18n.defaultText, dataIndex: 'count' },
      {
        title: commonUI.tables.cells.price.i18n.defaultText,
        dataIndex: 'price',
        render: (text: string, record: IRecord) => {
          const { price, currency } = record;
          return <Price price={price} currency={currency} />;
        },
      },
    ];

    return (
      <div className="root">
        <style jsx>{`
          .root :global(.ant-table-row:last-child td) {
            border: 0;
          }
        `}</style>
        <Table columns={columns} dataSource={this.getDataSource()} pagination={false} />
      </div>
    );
  }
}

export default CartItems;
