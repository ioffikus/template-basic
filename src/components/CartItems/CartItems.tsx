import React from 'react';
import { connect } from 'react-redux';
import { Table } from 'antd';
import cx from 'classnames';
import _ from 'lodash';
import Price from 'src/components/Price';
import CartProductImage from 'src/components/CartProductImage';
import Name from './Name';
import Amount from './Amount';
import Delete from './Delete';

interface IRecord {
  pictureUrl: string;
  nameFields: {
    name: string;
    shortDescription: string;
  };
  uid: string;
  key: string;
  count: number;
  cost: {
    currency: string;
    price: number;
  };
}

interface IOwnProps {
  mayDeleteItems?: boolean;
  mayChangeAmountItems?: boolean;
  footer?: React.ReactNode;
  cssModificationClass?: string;
  size?: 'large';
}

interface IConnectedState {
  ui: Alicanto.Models.UI;
  items: Alicanto.Models.CatalogItem[];
}

type IProps = IOwnProps & IConnectedState;

interface IState {}

const mapStateToProps = (state: Store.IState) => ({
  ui: state.ui,
  items: state.cart.items,
});

const WrappedCartItems = connect<IConnectedState, undefined, IOwnProps>(mapStateToProps)(
  class CartItems extends React.Component<IProps, IState> {
    getDataSource = (): IRecord[] => {
      return this.props.items.map(item => {
        const uid = item.item;
        const count = _.get(item, 'count', 0);
        const schemaFields = item.schema_fields;
        const pictureUrl = _.get(schemaFields, 'picture_url.value', '');
        const name = _.get(schemaFields, 'name.value', '');
        const shortDescription = _.get(schemaFields, 'short_description.value', '');
        const currency = _.get(schemaFields, 'currency.value', '');
        const price = _.get(schemaFields, 'price.value', 0);
        return {
          pictureUrl,
          count,
          uid,
          cost: { price, currency },
          nameFields: { shortDescription, name },
          key: uid,
        };
      });
    };

    render() {
      const { mayChangeAmountItems, mayDeleteItems, cssModificationClass, size, items, footer } = this.props;
      const cartUI = this.props.ui.routes.cart;
      const columns = [
        {
          key: 'image',
          className: 'image-col',
          render: (text: string, record: IRecord) => {
            const { pictureUrl, nameFields: { name } } = record;
            return <CartProductImage pictureUrl={pictureUrl} imageAlt={name} />;
          },
        },
        {
          key: 'name',
          className: 'name-col',
          render: (text: string, record: IRecord) => {
            const { uid, nameFields: { name, shortDescription } } = record;
            return <Name uid={uid} name={name} shortDescription={shortDescription} />;
          },
        },
        {
          key: 'count',
          className: 'count-col',
          render: (text: string, record: IRecord) => {
            const { uid, count } = record;
            return <Amount mayChangeAmountItems={mayChangeAmountItems} uid={uid} count={count} />;
          },
        },
        {
          key: 'cost',
          className: 'cost-col',
          render: (text: string, record: IRecord) => {
            const { price, currency } = record.cost;
            return <Price price={price} currency={currency} />;
          },
        },
        {
          key: 'delete',
          className: 'delete-col',
          render: (text: string, record: IRecord) => {
            const { uid } = record;
            return <Delete mayDeleteItems={mayDeleteItems} uid={uid} />;
          },
        },
      ];
      return (
        <div
          className={cx('root', {
            [cssModificationClass]: !!cssModificationClass,
            large: size === 'large',
          })}
        >
          <style jsx>{`
            .root {
              width: 540px;
              max-width: 100%;
              font-size: 14px;
              font-weight: bold;
              padding: 15px 0;
            }
            .large {
              width: 100%;
              padding-bottom: 0;
            }
            .root :global(.list) {
              margin-bottom: 16px;
            }
            .root :global(.name-col) {
              width: 100%;
              max-width: 0;
            }
            .root :global(.image-col) {
              padding-left: 0;
            }
            .root :global(.delete-col) {
              padding-right: 0;
            }
            .root :global(.cost-col) {
              text-align: right;
              white-space: nowrap;
            }
            .large :global(.cost-col) {
              padding-left: 30px;
            }
            .root :global(.ant-row-flex:not(:last-of-type)) {
              margin-bottom: 9px;
              padding-bottom: 8px;
            }
            .root :global(.button-wrapper) {
              text-align: right;
            }
            .root :global(.button) {
              min-width: 192px;
              letter-spacing: 2px;
              text-transform: uppercase;
              font-weight: 500;
              font-size: 11px;
              line-height: 32px;
              text-decoration: none;
            }
            .root :global(.button + .button) {
              margin-left: 12px;
            }
            .has-border-bottom {
              border-bottom: 2px solid #e5ecf4;
            }
            .has-border-bottom :global(.ant-table-tbody > tr:not(:last-of-type) > td) {
              border-bottom: 1px solid #e5ecf4;
            }
            .root :global(.ant-table-tbody > tr > td) {
              padding-top: 9px;
              padding-bottom: 9px;
              border: 0;
            }
            .root :global(.ant-table-tbody > tr:hover > td) {
              background: transparent;
            }
          `}</style>
          {_.isEmpty(items) ? (
            <div className="app-text-empty">{cartUI.emptyStatus.i18n.defaultText}</div>
          ) : (
            <div className="inner">
              <div className="list">
                <Table columns={columns} dataSource={this.getDataSource()} pagination={false} showHeader={false} />
              </div>
              {footer && <div className="button-wrapper">{footer}</div>}
            </div>
          )}
        </div>
      );
    }
  },
);

export default WrappedCartItems;
