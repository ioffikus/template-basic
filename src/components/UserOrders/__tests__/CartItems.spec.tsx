import React from 'react';
import { shallow } from 'enzyme';
import uiStateMock from 'src/store/__mocks__/uiStore';
import userOrdersStateMock from 'src/store/__mocks__/userOrdersStore';
import CartItems from '../CartItems';

describe(null, () => {
  it('render() with items', () => {
    const items = userOrdersStateMock.results[0].cart.items;
    const wrapper = shallow(<CartItems ui={uiStateMock} items={items} />);
    expect(wrapper).toMatchSnapshot();
  });
  it('render() without items', () => {
    const items: Alicanto.Models.CatalogItem[] = [];
    const wrapper = shallow(<CartItems ui={uiStateMock} items={items} />);
    expect(wrapper).toMatchSnapshot();
  });
});
