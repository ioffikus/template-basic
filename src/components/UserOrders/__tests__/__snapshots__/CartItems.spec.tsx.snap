// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render() with items 1`] = `
<div
  className="jsx-256313609 root"
>
  <JSXStyle
    css=".root.jsx-256313609 .ant-table-row:last-child td{border:0;}"
    styleId="256313609"
  />
  <Table
    bordered={false}
    className=""
    columns={
      Array [
        Object {
          "key": "image",
          "render": [Function],
          "title": "image",
        },
        Object {
          "key": "name",
          "render": [Function],
          "title": "name",
        },
        Object {
          "dataIndex": "count",
          "title": "count",
        },
        Object {
          "dataIndex": "price",
          "render": [Function],
          "title": "price",
        },
      ]
    }
    dataSource={
      Array [
        Object {
          "count": 2,
          "currency": "$",
          "key": 83,
          "name": "Sword of Destiny (The Witcher)",
          "pictureUrl": "http://localhost:8000/media/item_pictures/2017-12-14T140837.2674370000dc74b408d40b4b7783ee3232c6554960.jpg",
          "price": 0.01,
          "uid": "3cf12d0021c3438c9d43",
        },
      ]
    }
    indentSize={20}
    loading={false}
    locale={Object {}}
    pagination={false}
    prefixCls="ant-table"
    rowKey="key"
    rowSelection={null}
    showHeader={true}
    size="large"
    useFixedHeader={false}
  />
</div>
`;

exports[`render() without items 1`] = `
<div
  className="jsx-256313609 root"
>
  <JSXStyle
    css=".root.jsx-256313609 .ant-table-row:last-child td{border:0;}"
    styleId="256313609"
  />
  <Table
    bordered={false}
    className=""
    columns={
      Array [
        Object {
          "key": "image",
          "render": [Function],
          "title": "image",
        },
        Object {
          "key": "name",
          "render": [Function],
          "title": "name",
        },
        Object {
          "dataIndex": "count",
          "title": "count",
        },
        Object {
          "dataIndex": "price",
          "render": [Function],
          "title": "price",
        },
      ]
    }
    dataSource={Array []}
    indentSize={20}
    loading={false}
    locale={Object {}}
    pagination={false}
    prefixCls="ant-table"
    rowKey="key"
    rowSelection={null}
    showHeader={true}
    size="large"
    useFixedHeader={false}
  />
</div>
`;
